### PR TITLE
Feature/i18n/refactoring and tests

### PR DESCRIPTION
**Pull request by StyleT**

### CODE REVIEW

File: ilc/client.js [hunk 1]
```diff
@@ -5,38 +5,56 @@ import Router from './client/ClientRouter';
...
-const router = new Router(registryConf, state, singleSpa);
+
+const appErrorHandlerFactory = (appName, slotName) => {
+    return fragmentErrorHandlerFactory(registryConf, router.getCurrentRoute, appName, slotName);
+};
+
+const i18n = registryConf.settings.i18n.enabled
+    ? new I18n(registryConf.settings.i18n, singleSpa, appErrorHandlerFactory)
+    : null;
+const router = new Router(registryConf, state, i18n ? i18n.unlocalizeUrl : undefined, singleSpa);
```
Comment (nightnei) [inline ilc/client.js:34]: it's not critical, but will be better to put `undefined` arguments as last argument or as optional inside object. and in this case we won't set `undefiend` here.
https://github.com/namecheap/ilc/pull/230/files#diff-318e72ee3c278cb96c1bb2130002e8a9bcdebd2042222be2a5284e589a1bb75eR204

File: ilc/client/dispatchSynchronizedEvent.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    handlers.forEach(v => {
+        v.errorHandler = (...args) => errorHandler(v.actorId, ...args);
+        v.prepare = wrapWithPromise(v.prepare);
+        v.execute = wrapWithPromise(v.execute);
+    });
+
+    const preparationPromises = handlers.map(v => v.prepare(eventDetail));
+
+    return Promise.allSettled(preparationPromises).then(results => {
+        for (let ii = 0; ii < results.length; ii++) {
```
Comment (nightnei) [inline ilc/client/dispatchSynchronizedEvent.js:29]: why `ii`? 😁

File: ilc/client/dispatchSynchronizedEvent.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                ii--;
+            }
+        }
+
+        const executionPromises = handlers.map((v, i) => v.execute(eventDetail, results[i].value));
+
+        return Promise.allSettled(executionPromises);
+    }).then(results => {
+        results
+            .map((v, i) => {
```
Comment (nightnei) [inline ilc/client/dispatchSynchronizedEvent.js:45]: it's better to use `reduce` here

File: ilc/tests/helpers.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+const _ = require('lodash');
```
Comment (nightnei) [inline ilc/tests/helpers.js:1]: I propose to not use lodash if we can do it.
for example, here we can use `deepmerge` package.

**Review by nightnei — APPROVED** [review]

**Review by oleh-momot — APPROVED** [review]

**Pull request merged**